apps/api/pages/api/destination-calendars/[id]/_patch.ts

```diff
@@ -57,13 +60,93 @@ import { schemaQueryIdParseInt } from "~/lib/validations/shared/queryIdTransform
  *        description: Destination calendar not found
  */
 export async function patchHandler(req: NextApiRequest) {
-  const { prisma, query, body } = req;
+  const { userId, isAdmin, prisma, query, body } = req;
   const { id } = schemaQueryIdParseInt.parse(query);
   const parsedBody = schemaDestinationCalendarEditBodyParams.parse(body);
+  let credentialId: number | undefined = undefined;
+  const assignedUserId = isAdmin ? parsedBody.userId || userId : userId;
+
+  let queryConditions = { userId: assignedUserId };
+
+  // when linked with eventTypeId, we need to fetch the userId from the eventTypeId
+  if (parsedBody.eventTypeId) {
+    const eventType = await prisma.eventType.findFirst({
+      where: { id: parsedBody.eventTypeId, userId: assignedUserId },
+    });
+    if (!eventType)
+      throw new HttpError({
+        statusCode: 400,
+        message: "Bad request, eventTypeId invalid",
+      });
+    parsedBody.userId = null;
+    queryConditions = { eventTypeId: parsedBody.eventTypeId };
+  }
+
+  const destinationCalendarObject = await prisma.destinationCalendar.findFirst({
+    where: { id, ...queryConditions },
+    select: { credentialId: true },
+  });
```

Comment:
We query the destinationCalendar to fetch credentialId with given query conditions (userId or eventTypeId)